packages/library/src/components/bal-select/bal-select.tsx

```diff
@@ -218,300 +189,241 @@ export class Select {
-          </bal-dropdown-trigger>
-          <bal-dropdown-menu scrollable={this.scrollable}>
-            {this.renderFilter()}
-            <slot></slot>
-          </bal-dropdown-menu>
-        </bal-dropdown>
-      </Host>
-    )
   }
 
-  renderFilter() {
-    if (!this.multiple || !this.typeahead) {
-      return ''
+  private selectedFocusedOption() {
+    const visibleOptions = this.getVisibleOptions()
+    if (visibleOptions.length > this.focusIndex) {
+      const focusedOption = visibleOptions[this.focusIndex]
+      if (!isNil(focusedOption)) {
+        this.optionSelected({
+          id: focusedOption.id,
+          label: focusedOption.label || '',
+          value: focusedOption.value || '',
+        })
+      }
     }
-
-    return (
-      <div class="multiple-typeahead">
-        <div class="control has-icons-right">
-          <input
-            class="input"
-            autoComplete="off"
-            placeholder={this.filterPlaceholder}
-            onInput={this.onInput}
-            onKeyPress={this.onKeyPress}
-            onKeyUp={this.onKeyUp}
-            onBlur={e => this.balBlur.emit(e)}
-            onFocus={e => this.balFocus.emit(e)}
-            ref={el => (this.inputFilterElement = el as HTMLInputElement)}
-          />
-          <bal-icon class={{ 'is-hidden': this.loading }} turn={!this.loading && !this.typeahead && this.isDropdownOpen} size="small" color="info" name="search" />
-        </div>
-      </div>
-    )
   }
 
-  private async updateOptionProps() {
-    const inputValue = this.multiple && this.typeahead ? this.inputFilterElementValue : this.inputElementValue
-    this.clearFocus()
-    this.childOptions.forEach((option, index) => {
-      if (!this.noFilter && this.typeahead) {
-        const didMatch = this.compareForFiltering(`${option.label}` || '', `${inputValue}`)
-        option.setAttribute('hidden', `${!didMatch}`)
-      } else {
-        option.setAttribute('hidden', `${!this.isDropdownOpen}`)
+  private navigateWithArrowKey(event: KeyboardEvent) {
+    if (isArrowDownKey(event)) {
+      this.focusIndex = this.focusIndex + 1
+    } else {
+      if (isArrowUpKey(event)) {
+        this.focusIndex = this.focusIndex - 1
       }
-
-      const isSelected = !!this.value && this.value.includes(option.value as string)
-      option.setAttribute('selected', `${isSelected}`)
-
-      const isFocused = this.focusIndex === index
-      option.setAttribute('focused', `${isFocused}`)
-    })
-
-    const visibleOptions = this.childOptions.filter(o => !o.hidden)
-    if (visibleOptions.length === 1) {
-      this.clearFocus()
-      visibleOptions[0].setAttribute('focused', `${true}`)
     }
+  }
 
-    this.scrollToFocusedOption()
+  private focusOptionByLabel(key: string) {
+    this.labelToScrollTo = this.labelToScrollTo + key
+    clearTimeout(this.clearScrollToValue)
+    this.clearScrollToValue = setTimeout(() => {
+      this.scrollToLabel(this.labelToScrollTo)
+    }, 600)
```

Comment:
can this be done quicker or without a timeout?

Reply:
no this is the delay for the user to finish his search value